core/src/components/input/test/basic/input.e2e.ts

```diff
@@ -131,3 +131,82 @@ configs({ modes: ['ios'], directions: ['ltr'] }).forEach(({ title, screenshot, c
     });
   });
 });
+
+configs({ modes: ['ionic-md'] }).forEach(({ title, screenshot, config }) => {
+  test.describe(title('input: clear button in ionic theme, visual checks'), () => {
+    test('should not have visual regressions', async ({ page }) => {
+      await page.setContent(
+        `
+          <ion-input
+            label="Label"
+            label-placement="stacked"
+            clear-input="true"
+            value="Text"
+          ></ion-input>
+        `,
+        config
+      );
+
+      const input = page.locator('ion-input');
+      await input.evaluate((el: HTMLIonInputElement) => el.setFocus());
+      await page.waitForChanges();
+
+      await expect(input).toHaveScreenshot(screenshot(`input-with-clear-button`));
+    });
+
+    test('should not have visual regressions when clear button is focused', async ({ page }) => {
+      await page.setContent(
+        `
+          <ion-input
+            label="Label"
+            label-placement="stacked"
+            clear-input="true"
+            value="Text"
+          ></ion-input>
+        `,
+        config
+      );
+
+      const input = page.locator('ion-input');
+
+      await input.evaluate((el: HTMLIonInputElement) => el.setFocus());
+      await page.waitForChanges();
+
+      await page.keyboard.press('Tab');
+      await page.waitForChanges();
```

Comment:
The tests aren't capturing the focus state because `ion-focused` is never being added. It seems that we have to manually add that class based on other examples that depend on that class.

```suggestion
      const clearButton = input.locator('.input-clear-icon');
      clearButton.evaluate((el: HTMLElement) => el.classList.add('ion-focused'));
      await page.waitForChanges();
```

Reply:
Oh, thank you! That makes sense; the focus utility only works with `ion-app`. Saved me some stress this morning 👀

Reply:
Fixed in https://github.com/ionic-team/ionic-framework/pull/29337/commits/2e2ef4d77573c4a0a780d99297e0582e40050fed and https://github.com/ionic-team/ionic-framework/pull/29337/commits/f0a1f15fe4f594184a53950f80eff683e21adc0e 👍